### ISSUE TITLE
Latest Master nix-build Seg Fault on Docker Image Build

### ISSUE THREAD

**Issue by mbwmbw1337** [kadena-io/chainweb-data#189]
Having this across over 40 different nodes on various flavours of Ubuntu OS... Never had an issue building the docker image before.
```
trace: WARNING: 9.6.3 is out of date, consider using 9.6.4.
these 3 derivations will be built:
  /nix/store/qm10dzfq8fb7wmwsbz5qr4k0grn15yl1-docker-layer-chainweb-data.drv
  /nix/store/pihjkyv052jhr1c6694w966sl8anbyyl-runtime-deps.drv
  /nix/store/n3482wsdrfcc1d6nxrpdg01s1snbdgr6-docker-image-chainweb-data.tar.gz.drv
building '/nix/store/qm10dzfq8fb7wmwsbz5qr4k0grn15yl1-docker-layer-chainweb-data.drv'...
Formatting './image/disk-image.qcow2', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=1073741824 lazy_refcounts=off refcount_bits=16
Could not access KVM kernel module: Permission denied
qemu-kvm: failed to initialize kvm: Permission denied
qemu-kvm: falling back to tcg
cSeaBIOS (version rel-1.16.2-0-gea1b7a073390-prebuilt.qemu.org)


iPXE (http://ipxe.org) 00:03.0 CA00 PCI2.10 PnP PMM+1EFD0F00+1EF30F00 CA00
                                                                               


Booting from ROM...
Probing EDD (edd=off to disable)... ok
c./run-vm: line 5:    12 Segmentation fault      /nix/store/cfvrcbmb5bvvrz63f4c5dyvvijwfgg74-qemu-host-cpu-only-8.1.0/bin/qemu-kvm -cpu max -nographic -no-reboot -device virtio-rng-pci -virtfs local,path=/nix/store,security_model=none,mount_tag=store -virtfs local,path=/build/xchg,security_model=none,mount_tag=xchg -drive file=./image/disk-image.qcow2,if=virtio,cache=unsafe,werror=report -kernel /nix/store/ng20x62170qvivq1ym2gbcxq1s8vpl1d-linux-6.1.53/bzImage -initrd /nix/store/ijp66m0xyb3b9h998w3i7c9in9invw72-initrd/initrd -append "console=ttyS0 panic=1 command=/nix/store/0qgga5n54adny309857n7lwaz90hnbg1-vm-run-stage2 out=/nix/store/sb1lqgm3v38xnfn1l38z69z7r61f4l4r-docker-layer-chainweb-data mountDisk= loglevel=4" -m 512 -smp cpus=16
error: builder for '/nix/store/qm10dzfq8fb7wmwsbz5qr4k0grn15yl1-docker-layer-chainweb-data.drv' failed with exit code 139
error: 1 dependencies of derivation '/nix/store/n3482wsdrfcc1d6nxrpdg01s1snbdgr6-docker-image-chainweb-data.tar.gz.drv' failed to build
```

**Comment by enobayram** [comment]
Thank you for reporting this issue. Unfortunately I can't reproduce this locally, what version of Ubuntu is this happening on?

In any case, I've realized that the docker image builds for chainweb-data are simple enough that we can avoid relying on KVM entirely during the build of the image. I have made the necessary changes in https://github.com/kadena-io/chainweb-data/pull/190, would you mind checking if that branch works for you:
```
docker load < $(nix build github:kadena-io/chainweb-data#chainweb-data-docker --no-link --print-out-paths)
docker run -ti --rm chainweb-data:latest --help
```

**Comment by mbwmbw1337** [comment]
Built successfully with no errors. I am testing from a clean build as well and will update you shortly.

To be clear, to reproduce the error, I cloned the source and then ran `nix-build` in the source directory.

**Linux Version**
```
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy

```

**Issue with tcg - not qemu-kvm **
Note: I believe this was occurring because of the fallback from `qemu-kvm` to `tcg`. The build worked fine with `tcg` before but then failed. Once I added the nix-build users to the kvm and libvert groups, the build succeeded using `qemu-kvm`. Essentially you should only see the failure if your build account does not have access to `qemu-kvm` (either no permissions or `qemu-kvm` not installed). If you want to test the `tcg` build you need to disable kvm and the build will automatically fall back to  `tcg`.

**Comment by enobayram** [comment]
Thanks a lot for the deeper analysis, I guess the `dockerTools.buildImage` that comes with the `nixpkgs` version pinned [in this commit](https://github.com/kadena-io/chainweb-data/commit/8cefb143c8d296f1129dd3e2e08b61c46cf403d7#diff-216b2b7bfde9416c79d133bacb031e95702a20bdedb548c0b055c837aa4f6a9cL242-L257) changed something with the `tcg` fallback, which is good to know. I think, for now, it's best to avoid the issue entirely with the change in #190 as it requires no virtualization. I can merge that PR if it works for your setup and then we can close this issue.

**Comment by mbwmbw1337** [comment]
@enobayram  - this has been confirmed to fix the issue, including on a previously broken machine with no kvm enabled. Thanks again, I believe #190 is perfect.

**Comment by mbwmbw1337** [comment]
Solved by #190.

**Comment by enobayram** [comment]
Thank you for confirming!
